Libraries/LibWeb/CSS/Interpolation.cpp

```diff
@@ -673,8 +678,73 @@ NonnullRefPtr<CSSStyleValue const> interpolate_value(DOM::Element& element, Calc
         return StyleValueList::create(move(interpolated_values), from_list.separator());
     }
     default:
+        if (allow_discrete == AllowDiscrete::No)
+            return {};
         return delta >= 0.5f ? to : from;
     }
 }
 
+NonnullRefPtr<CSSStyleValue const> interpolate_value(DOM::Element& element, CalculationContext const& calculation_context, CSSStyleValue const& from, CSSStyleValue const& to, float delta)
+{
+    return *interpolate_value_impl(element, calculation_context, from, to, delta, AllowDiscrete::Yes);
+}
+
+NonnullRefPtr<CSSStyleValue const> interpolate_repeatable_list(DOM::Element& element, CalculationContext const& calculation_context, CSSStyleValue const& from, CSSStyleValue const& to, float delta)
+{
+    // https://www.w3.org/TR/web-animations/#repeatable-list
+    // Same as by computed value except that if the two lists have differing numbers of items, they are first repeated to the least common multiple number of items.
+    // Each item is then combined by computed value.
+    // If a pair of values cannot be combined or if any component value uses discrete animation, then the property values combine as discrete.
+
+    auto make_repeatable_list = [&](auto const& from_list, auto const& to_list, Function<void(NonnullRefPtr<CSSStyleValue const>)> append_callback) -> bool {
+        // If the number of components or the types of corresponding components do not match,
+        // or if any component value uses discrete animation and the two corresponding values do not match,
+        // then the property values combine as discrete
+        size_t from_index = 0;
+        size_t from_count = 0;
+        size_t to_index = 0;
+        size_t to_count = 0;
+
+        while (from_count == 0 || to_count == 0 || from_count != to_count) {
+            auto value = interpolate_value_impl(element, calculation_context, from_list.values()[from_index], to_list.values()[to_index], delta, AllowDiscrete::No);
+            if (!value)
+                return false;
+            append_callback(*value);
+            from_index++;
+            to_index++;
+            if (from_index == from_list.size()) {
+                from_count += from_list.size();
+                from_index = 0;
+            }
+            if (to_index == to_list.size()) {
+                to_count += to_list.size();
+                to_index = 0;
+            }
```

Comment:
FYI: `StyleValueList::value_at()` has an `allow_loop` parameter which does the wrapping automatically. That might make the logic here a bit simpler: figure out the lowest common multiple, and then just loop from 0 to that.

Reply:
My latest push adds a `gcd` and `lcm` implementation to AK, which allows me to do it this way instead.